warp10/src/main/java/io/warp10/script/functions/FETCH.java

```diff
@@ -428,13 +433,62 @@ public Object apply(WarpScriptStack stack) throws WarpScriptException {
         m.setLabelsId(GTSHelper.labelsId(this.SIPHASH_LABELS, m.getLabels()));
       }
 
+      //
+      // Now apply the scoping of the token if set
+      //
+
+      if (rtoken.getAttributesSize() > 0 && rtoken.getAttributes().containsKey(Constants.TOKEN_ATTR_SCOPE)) {
+
+        Object[] clslbls;
+        try {
+          clslbls = PARSESELECTOR.parse(rtoken.getAttributes().get(Constants.TOKEN_ATTR_SCOPE));
+        } catch (WarpScriptException e) {
+          throw new WarpScriptException("Invalid syntax for token scope selector.");
+        }
+
+        DirectoryRequest dr = new DirectoryRequest();
+
+        dr.addToClassSelectors((String) clslbls[0]);
+
+        MetadataIterator miter = new MetadataIterator() {
+          int idx = 0;
+
+          @Override
+          public boolean hasNext() {
+            return idx < modifiedMetas.size();
+          }
+
+          @Override
+          public Metadata next() {
+            if (idx >= modifiedMetas.size()) {
+              throw new IllegalStateException();
+            } else {
+              return modifiedMetas.get(idx++);
+            }
+          }
+
+          @Override
+          public void close() throws Exception {}
+        };
+
+        MetadataIterator mi = MetadataSelectorMatcher.getIterator(miter, dr);
```

Comment:
Null pointer exception here.
```
    {
      "type": "class java.lang.NullPointerException",
      "message": "Cannot invoke \"java.util.List.get(int)\" because the return value of \"io.warp10.continuum.store.thrift.data.DirectoryRequest.getLabelsSelectors()\" is null",
      "stacktrace": [
        [
          "MetadataSelectorMatcher.java",
          278,
          "io.warp10.continuum.gts.MetadataSelectorMatcher",
          "getIterator"
        ],
        [
          "FETCH.java",
          474,
          "io.warp10.script.functions.FETCH",
          "apply"
        ],
```